frame/support/src/storage/migration.rs

```diff
@@ -195,3 +197,154 @@ pub fn take_storage_item<K: Encode + Sized, T: Decode + Sized, H: StorageHasher>
 ) -> Option<T> {
 	take_storage_value(module, item, key.using_encoded(H::hash).as_ref())
 }
+
+/// Move a storage from a pallet prefix to another pallet prefix.
+///
+/// NOTE: It actually moves every key after the storage prefix to the new storage prefix,
+/// key on storage prefix is also moved.
+pub fn move_storage_from_pallet(item: &[u8], old_pallet: &[u8], new_pallet: &[u8]) {
+	let mut new_prefix = Vec::new();
+	new_prefix.extend_from_slice(&Twox128::hash(new_pallet));
```

Comment:
can we reuse anything from `StorageValue` struct to generate this key? I am worried that we have hardcoded this pattern of `twox ++ twox` in many places. External teams have probably made this mistake even more.

Reply:
unfortunately I made the generator traits StorageValue, StorageMap and StorageDoubleMap abstract on the key generation with a default implementation :-/ so we can't make use of it directly.
We can still introduce a function for generating this key though.

Also the end I think we can refactor quite a bit if we get rid of decl_storage.